crates/fe/src/task/compile.rs

```diff
@@ -0,0 +1,268 @@
+use std::fs;
+use std::io::{Error, Write};
+use std::path::Path;
+
+use clap::{ArgEnum, Args};
+use fe_common::diagnostics::print_diagnostics;
+use fe_common::files::SourceFileId;
+use fe_driver::CompiledModule;
+
+const DEFAULT_OUTPUT_DIR_NAME: &str = "output";
+const DEFAULT_INGOT: &str = "main";
+
+use super::utils::load_files_from_dir;
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ArgEnum, Debug)]
+enum Emit {
+    Abi,
+    Ast,
+    LoweredAst,
+    Bytecode,
+    Tokens,
+    Yul,
+}
+
+#[derive(Args)]
+#[clap(about = "Compile fe project or single file")]
```

Comment:
"Compile the current project" (assuming it defaults to the current project if no path is specified)